packages/dds/merge-tree/src/attributionCollection.ts

```diff
@@ -295,9 +301,11 @@ export class AttributionCollection implements IAttributionCollection<Attribution
 			// TODO Non null asserting, why is this not null?
 			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
 			splitCollection.offsets.push(Math.max(this.offsets[i]! - pos, 0));
-			// Non null asserting, this.keys[i] will always exist since we are looping though the length of it?
-			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-			splitCollection.keys.push(this.keys[i]!);
+			if (this.keys[i] !== undefined) {
+				// Non null asserting, this.keys[i] will always exist since we are looping though the length of it?
+				// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+				splitCollection.keys.push(this.keys[i]!);
```

Comment:
If you assigned `this.keys[i]` to a local variable before the undefined check, does the need for this assertion go away? That might be cleaner.